client/state/comments/test/selectors.js

```diff
@@ -2,77 +2,56 @@
  * External dependencies
  */
 import { expect } from 'chai';
-import Immutable from 'immutable';
 
 /***
  * Internal dependencies
  */
 import {
-	getCommentParentKey
-} from '../utils';
-import {
 	getPostOldestCommentDate,
 	getPostMostRecentCommentDate,
 	getCommentLike
 } from '../selectors';
 
+const state = {
+	comments: {
+		items: {
+			'1-1': [
+					{ ID: 1, parent: false, date: '2016-01-31T10:07:18-08:00', i_like: true, like_count: 5 },
+					{ ID: 2, parent: false, date: '2016-01-29T10:07:18-08:00', i_like: false, like_count: 456 },
+					{ ID: 3, parent: { ID: 1 }, date: '2017-01-31T10:07:18-08:00', i_like: false, like_count: 0 },
+					{ ID: 4, parent: { ID: 2 }, date: '2015-01-29T10:07:18-08:00', i_like: false, like_count: 0 }
+			]
+		}
+	}
+};
+
 describe( 'selectors', () => {
 	describe( '#getPostMostRecentCommentDate()', () => {
-		it( 'should get most recent date', () => {
-			const commentItems = [
-				{ ID: 1, date: '2016-01-31T10:07:18-08:00' },
-				{ ID: 2, date: '2016-01-29T10:07:18-08:00' }
-			];
-
-			const res = getPostMostRecentCommentDate( {
-				comments: {
-					items: Immutable.fromJS( {
-						[ getCommentParentKey( 1, 1 ) ]: commentItems
-					} )
-				}
-			}, 1, 1 );
+		it( 'should get most recent date, ignoring children', () => {
+			const res = getPostMostRecentCommentDate( state, 1, 1 );
 
-			expect( res ).to.be.eql( new Date( commentItems[ 0 ].date ) );
+			expect( res ).to.be.eql( new Date( '2016-01-31T10:07:18-08:00' ) );
 		} );
 
 		it( 'should return undefined if no comment items', () => {
 			const res = getPostMostRecentCommentDate( {
-				comments: {
-					items: Immutable.fromJS( {
-						[ getCommentParentKey( 1, 1 ) ]: []
-					} )
-				}
+				comments: { items: { '1-1': [] } }
 			}, 1, 1 );
 
 			expect( res ).to.be.eql( undefined );
 		} );
 	} ); // end of getPostMostRecentCommentDate
 
 	describe( '#getPostOldestCommentDate()', () => {
-		it( 'should get earliest date', () => {
-			const commentItems = [
-				{ ID: 1, date: '2016-01-31T10:07:18-08:00' },
-				{ ID: 2, date: '2016-01-29T10:07:18-08:00' }
-			];
+		it( 'should get earliest date, ignoring children', () => {
```

Comment:
Fixed on 18bc439ab4db1a1d243a4901d426fcb5f5fec705 and 16af101525c5641b560e34c7862f10c7f0e52faf.